core/schema/module.graphqls

```diff
@@ -0,0 +1,57 @@
+scalar ModuleID
+
+extend type Query {
+  """
+  Load a module by ID, or create a new one if id is unset.
+  """
+  module(id: ModuleID): Module!
+
+  """
+  The module currently being served in the session, if any.
+  """
+  currentModule: Module
+}
+
+type Module {
+  id: ModuleID!
+
+  "The name of the module"
+  name: String!
+
+  "The doc string of the module, if any"
+  description: String
+
+  "Functions served by this module"
+  functions: [Function!]
+
+  "This module plus the given Function associated with it"
+  withFunction(id: FunctionID!): Module!
+
+  """
+  Serve a module's API in the current session.
+      Note: this can only be called once per session.
+      In the future, it could return a stream or service to remove the side effect.
+  """
+  serve(environment: [ModuleEnvironmentVariable!]): Void!
```

Comment:
Per [the conversation in discord](https://discord.com/channels/707636530424053791/1146498746432094208/1147281294112333844), I removed `workdir` from here.